src/Api.elm

```diff
@@ -10,37 +10,36 @@ module Api exposing
     , toUrl
     )
 
+import FullyQualifiedName as FQN exposing (FQN)
+import Hash exposing (Hash)
 import Http
 import Json.Decode as Decode
+import Perspective
+    exposing
+        ( CodebasePerspectiveParam(..)
+        , Perspective(..)
+        , PerspectiveParams(..)
+        )
 import Regex
 import Syntax
 import Url.Builder exposing (QueryParameter, absolute, int, string)
 
 
-type ApiBasePath
-    = ApiBasePath (List String)
+
+-- ENDPOINT -------------------------------------------------------------------
 
 
 type Endpoint
     = Endpoint (List String) (List QueryParameter)
 
 
-type ApiRequest a msg
-    = ApiRequest Endpoint (Decode.Decoder a) (Result Http.Error a -> msg)
+list : PerspectiveParams -> String -> Endpoint
+list perspectiveParams fqnOrHash =
```

Comment:
Various endpoints now take a `Perspective` or (as in the case of list) a `PerspectiveParams` to ensure data is fetched within the right context.

Reply:
(The list API does not actually support rootBranch and relativeTo right now, but is intended to. Sending the params doesn't do anything as they are just ignored).